uqcsbot/whatsdue.py

```diff
@@ -74,18 +74,36 @@ async def whatsdue(
             await interaction.edit_original_response(content=e.message)
             return
 
-        message = (
-            "_*WARNING:* Assessment information may vary/change/be entirely"
-            + " different! Use at your own discretion_\n> "
+        embed = discord.Embed(
+            title=f"What's Due: {', '.join(course_names)}",
+            url=asses_page,
+            description="*WARNING: Assessment information may vary/change/be entirely different! Use at your own discretion*",
         )
-        message += "\n> ".join(map(self.get_formatted_assessment_item, assessment))
+        if assessment:
+            for assessment_item in assessment:
+                course, task, due, weight = assessment_item
+                embed.add_field(
+                    name=course,
+                    value=f"`{weight}` {task} **({due})**",
+                    inline=False,
+                )
+        elif fulloutput:
+            embed.add_field(
+                name="",
+                value=f"No assessment items could be found",
+            )
+        else:
+            embed.add_field(
+                name="",
+                value=f"Nothing seems to be due soon",
+            )
+
         if not fulloutput:
-            message += (
-                "\n_Note: This may not be the full assessment list. Set fulloutput"
-                + "to True for the full list._"
+            embed.set_footer(
+                text="Note: This may not be the full assessment list. Set fulloutput to True for the full list."
```

Comment:
From a quasi-legal-bullshit standpoint, I hesitate to let the bot say (or imply) "this is a complete list of all assessment", because if it shits the bed, it might not be. (And bed-shitting is related to UQ's page keeping the same format, so we're relying on an external entity here.)

Like, it's a violently stupid scenario, but there's a degree of ass-covering to be expected in case someone is fool enough to rely on the bot for assessment details.

Reply:
Have a look at the latest commit to see if the wordings better. Feel free to suggest any adjustments.

Reply:
That wording's good enough for me - just as long as somewhere it's got words equivalent to "read the ECP for the final word on what's due".